### PR TITLE
[optimization](be) optimization for ColumnConst when writing mysql result

### PR DESCRIPTION
**Pull request by zclllyybb**
# Proposed changes

## Problem summary

optimize to avoid expanding of ColumnConst when writing mysql results.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [x] Is this PR support rollback (If NO, please explain WHY)

### CODE REVIEW

**Comment by zclllyybb** [comment]
run buildall

File: be/src/vec/sink/vmysql_result_writer.cpp [hunk 1]
```diff
@@ -605,107 +611,113 @@ Status VMysqlResultWriter<is_binary_format>::append_block(Block& input_block) {
     // convert one batch
     auto result = std::make_unique<TFetchDataResult>();
     for (int i = 0; status.ok() && i < _output_vexpr_ctxs.size(); ++i) {
-        auto column_ptr = block.get_by_position(i).column->convert_to_full_column_if_const();
+        const auto& [column_ptr, col_const] = unpack_if_const(block.get_by_position(i).column);
         auto type_ptr = block.get_by_position(i).type;
 
+        if (UNLIKELY(num_rows != block.get_by_position(i).column->size())) {
```
Comment (HappenLee) [inline be/src/vec/sink/vmysql_result_writer.cpp:617]: Better change to DCHECK

File: be/src/vec/sink/vmysql_result_writer.cpp [hunk 2]
```diff
@@ -129,7 +127,7 @@ Status VMysqlResultWriter<is_binary_format>::_add_one_column(
             }
```
Comment (HappenLee) [inline be/src/vec/sink/vmysql_result_writer.cpp:128]: `const auto col_indx = index_check_const(i, arg_const)`

**Comment by zclllyybb** [comment]
run buildall

**Comment by zclllyybb** [comment]
run p0

**Comment by zclllyybb** [comment]
run p0

**Comment by zclllyybb** [comment]
run buildall

**Comment by zclllyybb** [comment]
run p0

**Comment by zclllyybb** [comment]
run p0

**Comment by zclllyybb** [comment]
run buildall

**Comment by zclllyybb** [comment]
run buildall

**Comment by zclllyybb** [comment]
run buildall

**Comment by zclllyybb** [comment]
run buildall

**Comment by zclllyybb** [comment]
run build all

**Comment by zclllyybb** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Comment by amorynan** [comment]
@HappenLee  please merge it

**Pull request merged**